File: app/src/main/java/com/kazakago/cleanarchitecture/data/api/WeatherRetrofit.java
package com.kazakago.cleanarchitecture.data.api;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * WeatherRetrofit from livedoor
 * <p>
 * Created by tamura_k on 2016/06/03.
 */
public class WeatherRetrofit {

    public static Retrofit getInstance() {
        return new Retrofit.Builder()
                .baseUrl("http://weather.livedoor.com/forecast/webservice/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }
}

File: app/src/main/java/com/kazakago/cleanarchitecture/web/api/WeatherRetrofit.kt
package com.kazakago.cleanarchitecture.web.api

import com.squareup.moshi.KotlinJsonAdapterFactory
import com.squareup.moshi.Moshi
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.moshi.MoshiConverterFactory

object WeatherRetrofit {

    val instance: Retrofit
        get() {
            val moshi = Moshi.Builder()
                    .add(KotlinJsonAdapterFactory())
                    .build()
            return Retrofit.Builder()
                    .baseUrl("http://weather.livedoor.com/forecast/webservice/")
                    .addConverterFactory(MoshiConverterFactory.create(moshi))
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build()
        }

}
